f8a_worker/workers/recommender.py

```diff
@@ -607,6 +608,39 @@ def _get_stack_values(self, similar_stacks_list):
             similarity_list.append(s_stack)
         return similarity_list
 
+def apply_license_filter(epv_list):
+    dependencies = []
+    licenses = []
+    license_score_list = []
+    for epv in epv_list:
+        name = epv.get('pkg', {}).get('name', [''])[0]
+        version = epv.get('ver', {}).get('version', [''])[0]
+        licenses = epv.get('ver', {}).get('version', {}).get('licenses', [])
+
+        license_scoring_input = {
+            'package': name,
+            'version': version,
+            'licenses': licenses
+        }
+        license_score_list.append(license_scoring_input)
+
+    # Call License Scoring to Get Stack License
+    license_analysis, dependencies = perform_license_analysis(license_score_list, dependencies)
```

Comment:
this is really strange... you are using one variable `dependencies` as an argument of the call and then as the return value. as the argument, it's always `[]`...

Reply:
This was old logic. I have done a new logic that does not require it any more. I plan to close this PR and open a new PR for new logic.